src/test/groovy/com/synacy/gradprogram/spock/exercise/DeliveryServiceSpec.groovy

```diff
@@ -0,0 +1,50 @@
+package com.synacy.gradprogram.spock.exercise
+
+import spock.lang.Specification
+
+class DeliveryServiceSpec extends Specification {
+
+    DeliveryService service
+
+    DateUtils dateUtils = Mock(DateUtils)
+    DeliveryRequestRepository deliveryRequestRepository = Mock(DeliveryRequestRepository)
+
+    def setup() {
+        service = new DeliveryService(dateUtils, deliveryRequestRepository)
+    }
+
+    def "createDelivery should be able to save the created delivery request."(){
+        given:
+        Order order = new Order()
+
+        UUID id = order.getId()
+        DateUtils deliveryDate = dateUtils.getCurrentDate()
+        Courier courier = Courier.JRS
+
+        when:
+        service.createDelivery(order)
+
+        then:
+        1 * deliveryRequestRepository.saveDeliveryRequest(_) >> { DeliveryRequest deliveryRequest ->
+            assert id == deliveryRequest.getOrderId()
+            assert deliveryDate == deliveryRequest.getDeliveryDate()
```

Comment:
This is only passing because it is checking `null` against `null`. We should improve this.